core/src/main/java/org/mule/runtime/core/internal/util/generics/MethodParameter.java

```diff
@@ -329,7 +330,7 @@ public Integer getTypeIndexForLevel(int nestingLevel) {
    */
   private Map<Integer, Integer> getTypeIndexesPerLevel() {
     if (this.typeIndexesPerLevel == null) {
-      this.typeIndexesPerLevel = new HashMap<Integer, Integer>(4);
+      this.typeIndexesPerLevel = new SmallMap<>();
```

Comment:
`forSize(4)`

Reply:
actually no.. the key is an integer, this shouldn't be a smallmap